### ISSUE TITLE
Anchor components with and onClick but no href are skipped by keyboard navigation

### ISSUE THREAD

**Issue by danielm-hpe** [grommet/grommet#7634]
When using the Anchor component, browsers can decide whether to include them in keyboard navigation, and chrome for example utilizes the presence of an href prop to determine whether the anchor is actually interactive.  However, if the onclick is present without an href, it's interactive, but not reachable via keyboard navigation.  This creates an accessibility issue.

### Is this something you or your team is interested in contributing a pull request towards?
Not currently

### Expected Behavior
In the case where an Anchor has an onClick but no href, it should still be able to receive focus via keyboard navigation in all major browsers.

I'd probably expect it to be a button styled as an anchor.  I believe button has some convenience behavior where it can be rendered as an anchor, but not vice versa.

### Actual Behavior
Using tab / shift + tab when there's an Anchor without a href but with an onClick results in ignoring / skipping over the element.

### URL, screen shot, or Codepen exhibiting the issue
https://codesandbox.io/p/devbox/grommet-v2-template-forked-x62c7w?workspaceId=ws_BquGKHP9dxQiBsfUwDU9gV

### Steps to Reproduce

1. add an anchor without an href but with an onClick
2. try to reach it via keyboard navigation

### Your Environment

- Grommet version: 2.47.0
- Browser Name and version: chrome, firefox
- Operating System and version (desktop or mobile): windows 10
